ecmascript/transforms/module/src/amd.rs

```diff
@@ -22,18 +24,40 @@ pub fn amd(config: Config) -> impl Fold {
     Amd {
         config,
         in_top_level: Default::default(),
-        scope: Default::default(),
+        scope: RefCell::new(Default::default()),
         exports: Default::default(),
     }
 }
-
 struct Amd {
     config: Config,
     in_top_level: bool,
-    scope: Scope,
+    scope: RefCell<Scope>,
     exports: Exports,
 }
 
+impl Fold for Amd {
```

Comment:
Can you merge `Amd` and `AmdWorker`?
Creating a new visitor is  really bad for binary size.